e2e/cluster/e2e_test.go

```diff
@@ -0,0 +1,159 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build clusterE2E
+
+package cluster
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/cloudoperators/greenhouse/e2e/cluster/expect"
+	"github.com/cloudoperators/greenhouse/e2e/shared"
+	greenhousev1alpha1 "github.com/cloudoperators/greenhouse/pkg/apis/greenhouse/v1alpha1"
+	"github.com/cloudoperators/greenhouse/pkg/clientutil"
+)
+
+const (
+	remoteClusterName = "remote-int-cluster"
+)
+
+var (
+	env              *shared.TestEnv
+	ctx              context.Context
+	adminClient      client.Client
+	remoteClient     client.Client
+	remoteRestClient *clientutil.RestClientGetter
+	testStartTime    time.Time
+)
+
+func TestE2e(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Cluster E2E Suite")
+}
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
+	ctx = context.Background()
+	env = shared.NewExecutionEnv(greenhousev1alpha1.AddToScheme).WithOrganization(ctx, "./testdata/organization.yaml")
+	adminClient = env.GetClient(shared.AdminClient)
+	Expect(adminClient).ToNot(BeNil(), "admin client should not be nil")
+	remoteClient = env.GetClient(shared.RemoteClient)
+	Expect(remoteClient).ToNot(BeNil(), "remote client should not be nil")
+	remoteRestClient = env.GetRESTClient(shared.RemoteRESTClient)
+	Expect(remoteRestClient).ToNot(BeNil(), "remote rest client should not be nil")
```

Comment:
The dependency towards `clientutil` is already [there](https://github.com/cloudoperators/greenhouse/blob/7f31396e5bb7f0eeb4e81f35ba3a164eccc7f46f/e2e/shared/e2e.go#L188) and I think the e2e will probably use the same schemas that Greenhouse uses as well. 
I just want to avoid that we build similar functionality in two places of the codebase.

Reply:
ok done